frontend/lib/components/TextEditor/extensions/AIHighlight.ts

```diff
@@ -0,0 +1,61 @@
+import { Command } from "@tiptap/core";
+import { Highlight } from "@tiptap/extension-highlight";
+
+import { AIHighlightOptions } from "./types";
+
+const unsetSelectionsInDocument: Command = ({ dispatch, state }) => {
+  const { tr } = state;
+  if (!dispatch) {
+    return false;
+  }
+
+  state.doc.descendants((node, pos) => {
+    if (node.isText) {
+      node.marks.forEach((mark) => {
+        if (mark.type.name === "aiHighlight") {
+          if (mark.attrs["type"] !== "selection") {
+            return false;
+          }
+
+          tr.removeMark(pos, pos + node.nodeSize, mark.type);
+        }
+      });
+    }
+  });
+
+  dispatch(tr);
```

Comment:
can you be more explicit in variable names here ? :)

Reply:
```suggestion
state.doc.descendants((node, pos) => {
  if (node.isText) {
    node.marks.some((mark) => 
      mark.type.name === "aiHighlight" && mark.attrs["type"] === "selection" &&
      tr.removeMark(pos, pos + node.nodeSize, mark.type)
    );
  }
});
```

Reply:
This raises an eslint error
![image](https://github.com/user-attachments/assets/55b46e04-cbf1-4016-b8e8-30768c54ab5b)
Is this better?
```ts
  state.doc.descendants((node, pos) => {
    if (node.isText) {
      node.marks.map((mark) => {
        if (
          mark.type.name === AIHighlight.name &&
          mark.attrs["type"] === "selection"
        ) {
          tr.removeMark(pos, pos + node.nodeSize, mark.type);
        }
      });
    }
  });

```

Reply:
Yes perfect